sys/net/application_layer/gcoap/gcoap.c

```diff
@@ -570,6 +570,32 @@ static void _find_obs_memo_resource(gcoap_observe_memo_t **memo,
 }
 
 /*
+ * Check listener structure
+ *
+ *  - resources should be sorted alphabetically
+ */
+#ifdef DEVELHELP
+static void _check_listener(gcoap_listener_t *listener)
+{
+    const char *prev_path = "";
+    const char *path = NULL;
+
+    for (size_t i = 0; i < listener->resources_len; i++) {
```

Comment:
Why not start with `i = 1` and use `listener->resources[i-1]->path` and `listener->resources[i]->path` in `strcmp` that way omitting these extra variables plus remembering the previous path.